src/commands/utils/produceMessage.test.ts

```diff
@@ -1,89 +1,50 @@
 import * as assert from "assert";
-import * as sinon from "sinon";
-import { TEST_LOCAL_SCHEMA } from "../../../tests/unit/testResources";
-import { TEST_DIRECT_CONNECTION_FORM_SPEC } from "../../../tests/unit/testResources/connection";
-import { getTestExtensionContext } from "../../../tests/unit/testUtils";
+import { TEST_LOCAL_KEY_SCHEMA, TEST_LOCAL_SCHEMA } from "../../../tests/unit/testResources";
 import { Schema } from "../../models/schema";
 import { SchemaInfo, SubjectNameStrategy } from "../../schemas/produceMessageSchema";
-import { CustomConnectionSpec, getResourceManager } from "../../storage/resourceManager";
 import { createProduceRequestData, extractSchemaInfo } from "./produceMessage";
+import { ProduceMessageSchemaOptions } from "./types";
 
 describe("commands/utils/produceMessage.ts createProduceRequestData()", function () {
-  let sandbox: sinon.SinonSandbox;
-  let getDirectConnectionStub: sinon.SinonStub;
-
-  before(async function () {
-    await getTestExtensionContext();
-  });
-
-  beforeEach(function () {
-    sandbox = sinon.createSandbox();
-
-    getDirectConnectionStub = sandbox.stub(getResourceManager(), "getDirectConnection");
-  });
-
-  afterEach(function () {
-    sandbox.restore();
-  });
-
-  it("should create request data with 'type' set for CCloud topics", async function () {
-    const result = await createProduceRequestData(
-      {
-        key: "test-key",
-        value: "test-value",
-      },
-      {},
-      true,
-    );
-
-    assert.deepStrictEqual(result, {
-      keyData: {
-        type: "JSON",
-        data: "test-key",
-      },
-      valueData: {
-        type: "JSON",
-        data: "test-value",
-      },
-    });
-  });
-
-  it("should create request data without 'type' for local topics", async function () {
+  it("should nest key/value data under keyData.data and valueData.data", async function () {
     const result = await createProduceRequestData({
       key: "test-key",
       value: "test-value",
     });
 
     assert.deepStrictEqual(result, {
-      keyData: {
-        data: "test-key",
-      },
-      valueData: {
-        data: "test-value",
-      },
+      keyData: { data: "test-key" },
+      valueData: { data: "test-value" },
     });
   });
 
-  it("should create request data without 'type' for direct connection topics", async function () {
-    // even if it's a CCloud direct connection, we don't want to set 'type' since the sidecar will
-    // send it directly to the Kafka cluster instead of through the REST proxy
-    const fakeSpec: CustomConnectionSpec = {
```

Comment:
Also no longer applicable, since it was previously to check that we sent `type` for CCloud-flavored direct connections.